coderd/healthcheck/derp.go

```diff
@@ -181,60 +194,78 @@ func (r *DERPNodeReport) doExchangeMessage(ctx context.Context) {
 		return
 	}
 
-	var peerKey atomic.Pointer[key.NodePublic]
-	eg, ctx := errgroup.WithContext(ctx)
+	var (
+		peerKey  atomic.Pointer[key.NodePublic]
+		lastSent atomic.Pointer[time.Time]
+	)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	wg := &sync.WaitGroup{}
 
 	receive, receiveID, err := r.derpClient(ctx, r.derpURL())
 	if err != nil {
 		return
 	}
 	defer receive.Close()
 
-	eg.Go(func() error {
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
 		defer receive.Close()
 
 		pkt, err := r.recvData(receive)
 		if err != nil {
 			r.writeClientErr(receiveID, xerrors.Errorf("recv derp message: %w", err))
-			return err
+			return
 		}
 
 		if *peerKey.Load() != pkt.Source {
 			r.writeClientErr(receiveID, xerrors.Errorf("received pkt from unknown peer: %s", pkt.Source.ShortString()))
-			return err
+			return
 		}
 
-		t, err := time.Parse(time.RFC3339Nano, string(pkt.Data))
-		if err != nil {
-			r.writeClientErr(receiveID, xerrors.Errorf("parse time from peer: %w", err))
-			return err
-		}
+		t := lastSent.Load()
 
 		r.mu.Lock()
 		r.CanExchangeMessages = true
-		r.RoundTripPing = time.Since(t)
+		r.RoundTripPing = time.Since(*t)
```

Comment:
Since the send loop updates last sent every second, do we know that we're actually benchmarking the correct ping/ping?